patch_parser.py

```diff
@@ -348,12 +350,17 @@ def parse_comments(email_thread: Message) -> Patchset:
         comments = []
         for reply in patch.children:
             comments.extend(diff_reply(patch, reply))
-        set_index, length = parse_set_index(patch)
-        assert length == len(patches)
-        text = 'From: {from_}\nSubject: {subject}\n\n{content}'.format(
-                from_=patch.from_, subject=patch.subject, content=patch.content)
-        patch_list.append(Patch(text=patch.content, text_with_headers=text, set_index=set_index, comments=comments))
-        patch_list.sort(key=lambda x: x.set_index)
+        if (email_thread.in_reply_to):
+            set_index, length = parse_set_index(patch)
+            assert length == len(patches)
+            text = 'From: {from_}\nSubject: {subject}\n\n{content}'.format(
+                    from_=patch.from_, subject=patch.subject, content=patch.content)
+            patch_list.append(Patch(text=patch.content, text_with_headers=text, set_index=set_index, comments=comments))
+            patch_list.sort(key=lambda x: x.set_index)
+        else:
```

Comment:
seems like this branch only differs in terms of set_index.

Perhaps we can refactor this to

```py
# Patchsets with a single patch might not include "PATCH 1/1"
  if len(patches) == 1:
    set_index = 0
  else:
    set_index, length = parse_set_index(patch)
    assert length == len(patches)
```

Reply:
so I wasn't sure how complex everything was behind the scenes, which is why I used the else instead of just adjusting the set index and length. But, yeah I definitely agree this is the way to go.